src/dev.rs

```diff
@@ -640,44 +656,91 @@ impl<F: FieldExt> MockProver<F> {
+                        queries: &'a [(Column<T>, Rotation)],
+                        cells: &'a [Vec<F>],
+                    ) -> impl Fn(usize, usize, Rotation) -> Value<F> + 'a {
+                        move |index, _, _| {
+                            let (column, at) = &queries[index];
+                            let resolved_row = (row + at.0) % n;
+                            Value::Real(cells[column.index()][resolved_row as usize])
                         }
-
-                        fn load<'a, F: FieldExt, T: ColumnType>(
-                            n: i32,
-                            row: i32,
-                            queries: &'a [(Column<T>, Rotation)],
-                            cells: &'a [Vec<CellValue<F>>],
-                        ) -> impl Fn(usize, usize, Rotation) -> Value<F> + 'a
-                        {
-                            move |index, _, _| {
-                                let (column, at) = &queries[index];
-                                let resolved_row = (row + at.0) % n;
-                                cells[column.index()][resolved_row as usize].into()
+                    }
+
+                    fn load<'a, F: FieldExt, T: ColumnType>(
+                        n: i32,
+                        row: i32,
+                        queries: &'a [(Column<T>, Rotation)],
+                        cells: &'a [Vec<CellValue<F>>],
+                    ) -> impl Fn(usize, usize, Rotation) -> Value<F> + 'a {
+                        move |index, _, _| {
+                            let (column, at) = &queries[index];
+                            let resolved_row = (row + at.0) % n;
+                            cells[column.index()][resolved_row as usize].into()
+                        }
+                    }
+
+                    fn cell_value<'a, F: FieldExt>(
+                        virtual_cells: &'a [VirtualCell],
+                        column_type: Any,
+                        load: impl Fn(usize, usize, Rotation) -> Value<F> + 'a,
+                    ) -> impl Fn(usize, usize, Rotation) -> BTreeMap<metadata::VirtualCell, String> + 'a
+                    {
+                        let format_value = move |v: F| {
+                            if v.is_zero_vartime() {
+                                "0".into()
+                            } else if v == F::one() {
+                                "1".into()
+                            } else if v == -F::one() {
+                                "-1".into()
+                            } else {
+                                // Format value as hex.
+                                let s = format!("{:?}", v);
+                                // Remove leading zeroes.
+                                let s = s.strip_prefix("0x").unwrap();
+                                let s = s.trim_start_matches('0');
+                                format!("0x{}", s)
                             }
+                        };
```

Comment:
Can this be moved to a utility function, rather than being inlined in this already-very-convoluted method?